python/semantic_kernel/connectors/ai/function_choice_behavior.py

```diff
@@ -0,0 +1,281 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import logging
+from collections.abc import Callable
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Literal
+
+from pydantic.dataclasses import dataclass
+
+from semantic_kernel.exceptions.service_exceptions import ServiceInvalidExecutionSettingsError
+from semantic_kernel.functions.kernel_function_metadata import KernelFunctionMetadata
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+if TYPE_CHECKING:
+    from semantic_kernel.connectors.ai.function_call_behavior import FunctionCallBehavior
+    from semantic_kernel.connectors.ai.prompt_execution_settings import PromptExecutionSettings
+    from semantic_kernel.kernel import Kernel
+
+DEFAULT_MAX_AUTO_INVOKE_ATTEMPTS = 5
+
+logger = logging.getLogger(__name__)
+
+
+class FunctionChoiceType(Enum):
+    """The type of function choice behavior."""
+
+    AUTO = "auto"
+    NONE = "none"
+    REQUIRED = "required"
+
+
+def _check_for_missing_functions(function_names: list[str], kernel_function_metadata: set) -> None:
+    """Check for missing functions in the kernel function metadata."""
+    try:
+        qualified_names = {metadata.fully_qualified_name for metadata in kernel_function_metadata}
+        missing_functions = [name for name in function_names if name not in qualified_names]
+    except Exception:
+        logger.exception("Error checking for missing functions.")
+        return
+
+    if missing_functions:
+        raise ServiceInvalidExecutionSettingsError(
+            f"Unable to find the required function(s) `{', '.join(missing_functions)}` "
+            f"in the list of available kernel functions: {', '.join(qualified_names)}."
+        )
+
+
+@dataclass
+class FunctionCallChoiceConfiguration:
+    available_functions: list["KernelFunctionMetadata"] | None = None
```

Comment:
Do we still need this class?

Reply:
This is the specific class for `FunctionChoiceBehavior` so we will keep it. The old class name is `FunctionCallConfiguration`.